android/titanium/src/java/org/appcelerator/titanium/TiBlob.java

```diff
@@ -872,6 +872,60 @@ public TiBlob imageAsCompressed(Number compressionQuality)
 		return result;
 	}
 
+	@Kroll.method
+	public TiBlob imageAsWebP(Number compressionQuality)
+	{
+		Bitmap img = getImage();
+		if (img == null) {
+			return null;
+		}
+
+		float quality = 1f;
+		if (compressionQuality != null) {
+			quality = compressionQuality.floatValue();
+		}
+
+		TiBlob result = null;
+		ByteArrayOutputStream bos;
+
+		try {
+			bos = new ByteArrayOutputStream();
+			if (img.compress(
+				quality == 1f ? CompressFormat.WEBP_LOSSLESS : CompressFormat.WEBP_LOSSY, (int) (quality * 100), bos)
+			) {
+				byte[] data = bos.toByteArray();
```

Comment:
I would immediately try and free the image memory here before taking the byte array and generating a blob. Also, the whole segment feels very much like it should be part of `TiBlob.blobFromImage`. That's currently hard-coded to assume PNG/JPEG based on alpha, but probably should be refactored to have another variant where you explicitly specify the format and quality.

Otherwise, I'm not sure `width`/`height`/`uprightWidth`/`uprightHeight` will be set properly on the blob.